Bottom: d6b650e59e7fea872e5b95c37aa6f84ac8a2949b
Top:    4762dd0eddb07c97f23ad79a20a6e9ab96e78d6d
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-09-05 16:51:26 -0500

Refresh of add-haproxy-role

---
diff --git a/roles/haproxy/files/haproxy.cfg b/roles/haproxy/files/haproxy.cfg
index 5206b838..e4401689 100644
--- a/roles/haproxy/files/haproxy.cfg
+++ b/roles/haproxy/files/haproxy.cfg
@@ -1,23 +1,98 @@
 global
     daemon
-    maxconn 4096
+    maxconn 256
+    ssl-dh-param-file /etc/ssl/private/dhparam.pem
 
 defaults
     mode http
-    timeout connect 5000ms
-    timeout client 50000ms
-    timeout server 50000ms
+    timeout connect 5s
+    timeout client 50s
+    timeout server 50s
 
-frontend http-in
-    bind 0.0.0.0:80
-    acl is_nexus       hdr(host)       eq nexus.novatech-llc.com
-    acl is_buildbot    hdr(buildbot)   eq buildbot.novatech-llc.com
+#enable stats page
+listen stats
+    bind :9000
+    mode http
+    stats enable
+    stats hide-version
+    stats realm Haproxy\ Statistics
+    stats uri /haproxy_stats
+    stats auth stats:stats
+    stats refresh 5s
+
+#
+frontend container_https_in
+    mode http
+    bind *:80
+    bind *:443 ssl crt /etc/ssl/private/haproxy.pem
+
+    acl is_buildbot     hdr(host) eq buildbot.novatech-llc.com
+    acl is_gitlab       hdr(host) eq git.novatech-llc.com
+    acl is_ldapadmin    hdr(host) eq ldap.novatech-llc.com
+    acl is_mantisbt     hdr(host) eq mantis.novatech-llc.com
+    acl is_nexus        hdr(host) eq nexus.novatech-llc.com
+    acl is_svn          hdr(host) eq svn.novatech-llc.com
+    acl is_wiki         hdr(host) eq wiki.novatech-llc.com
+
+    use_backend backend-buildbot if is_buildbot
+    use_backend backend-nexus if is_nexus
+
+frontend container_ldap_in
+    mode tcp
+    bind *:389
+    bind *:636
+
+    use_backend backend-ldap
+
+frontend container_smtp_in
+    mode tcp
+    bind *:25
+    bind *:465
+    bind *:587
+
+    use_backend backend-exim4
+
+frontend container_ssh_in
+    mode tcp
+    bind *:22
+    default_backend
+
+#
+backend backend-svn
+    redirect scheme https code 301 if !{ ssl_fc }
+    server svn.frontend:80 check
+
+backend backend-phpldapadmin
+    redirect scheme https code 301 if !{ ssl_fc }
+    server phpldapadmin.frontend:80 check
+
+backend backend-exim4
+    server exim4.frontend:25 check
+
+backend backend-gitlab
+    redirect scheme https code 301 if !{ ssl_fc }
+    server gitlab.frontend:80 check
+
+backend backend-wiki
+    redirect scheme https code 301 if !{ ssl_fc }
+    server wiki.frontend:80 check
+
+backend backend-mantisbt
+    redirect scheme https code 301 if !{ ssl_fc }
+    server mantisbt.frontend:80 check
+
+backend backend-build
+    redirect scheme https code 301 if !{ ssl_fc }
+    server build.frontend:80 check
 
-    use_backend nexus if is_nexus
-    use_backend buildbot if is_buildbot
+backend backend-testssh
+    redirect scheme https code 301 if !{ ssl_fc }
+    server testssh.frontend:80 check
 
-backend buildbot
-    server buildbot1 buildbot.local:8080 maxconn 32
+backend backend-buildbot
+    redirect scheme https code 301 if !{ ssl_fc }
+    server buildbot.frontend:80 check
 
-backend nexus
-    server nexus1 nexus.local:8081 maxconn 32
+backend backend-nexus
+    redirect scheme https code 301 if !{ ssl_fc }
+    server nexus.frontend:80 check
